### PR TITLE
Fix #184, add build and test workflow

### LINKED ISSUE #184: Add functional test workflow
Add a github workflow similar to the functional test workflow for CFE/CFS that builds the software, executes functional tests, and checks the results.

Example for CFE is here: https://github.com/nasa/cFE/blob/be88a07313ff3b579c8abaef4b4a9085e74aabdf/.github/workflows/functional-tests.yml

A similar process can be done for bplib

### PR DESCRIPTION
**Pull request by jphickey**
**Describe the contribution**
Workflow to build and test BPLib in a variety of configurations.  The workflow implements a matrix of Debug and Release build types, and OSAL and POSIX adapter layers.

For the OSAL builds, the unit tests are also enabled, and the tests are executed.

Additionally, for the Debug OSAL build, the coverage statistics are computed and the results are included in the summary report, so they can be monitored.

This also migrates the "bptest" test program into a proper functional test based on UT Assert.

Fixes #184

**Testing performed**
Execute github workflow and check results

**Expected behavior changes**
The "bptest" debug program is removed, replaced with a (mostly equivalent) "sanity-test" program in the "ut-functional" directory.  The replacement is based on UT Assert, like other tests.

**System(s) tested on**
Ubuntu 22.04, Github Workflow hosted environments

**Contributor Info - All information REQUIRED for consideration of pull request**
Joseph Hickey, Vantage Systems, Inc.

### CODE REVIEW

**Review by glan6388 — APPROVED** [review]

**Comment by jphickey** [comment]
Discussed at 2022-11-09, will wait for review by @dzbaker next week.

**Comment by jphickey** [comment]
Reviewed at 2022-11-16 CCB, approved for merge

**Pull request merged**
